x-pack/solutions/security/packages/kbn-cloud-security-posture/public/src/components/fleet_extensions/azure_credentials_form/azure_credential_guides.test.tsx

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+import { render, screen } from '@testing-library/react';
+import { I18nProvider } from '@kbn/i18n-react';
+import { AzureSelectedCredentialsGuide } from './azure_credential_guides';
+import type { AzureCredentialsType } from '../types';
+import * as cloudSetupContext from '../hooks/use_cloud_setup_context';
+import { AZURE_CLOUD_CONNECTOR_SETUP_INSTRUCTIONS_TEST_SUBJ } from '@kbn/cloud-security-posture-common';
+
+// Mock the cloud setup context
+const mockUseCloudSetup = jest.fn();
+jest.spyOn(cloudSetupContext, 'useCloudSetup').mockImplementation(mockUseCloudSetup);
+
+const defaultMockCloudSetup = {
+  isAwsCloudConnectorEnabled: false,
+  isAzureCloudConnectorEnabled: true,
+  isGcpCloudConnectorEnabled: false,
+  isConfigContextLoading: false,
+  cloudContextData: undefined,
+  packageData: undefined,
+  packageInfo: {
+    name: 'cloud_security_posture',
+    version: '1.0.0',
+    title: 'Cloud Security Posture',
+  },
+  azurePolicyType: 'cloudbeat/cis_azure',
+  azureOrganizationEnabled: true,
+  shortName: 'CSPM',
+  elasticStackId: 'test-elastic-stack-id-12345',
+};
+
+const renderWithIntl = (component: React.ReactElement) =>
+  render(<I18nProvider>{component}</I18nProvider>);
+
+describe('AzureSelectedCredentialsGuide', () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+    mockUseCloudSetup.mockReturnValue(defaultMockCloudSetup);
+  });
+
+  describe('cloud_connectors credential type', () => {
+    const credentialType: AzureCredentialsType = 'cloud_connectors';
+
+    it('renders description and accordion with all setup steps', () => {
+      renderWithIntl(<AzureSelectedCredentialsGuide azureCredentialType={credentialType} />);
```

Comment:
nit not directly related to this PR: `AzureSelectedCredentialsGuide` the name of the component is quite generic why it is specifically the cloud connector guide, would make sense to name it to reflect that fact